src/components/PlayHistory/components/ToggleOrder.tsx

```diff
@@ -0,0 +1,27 @@
+import { useState } from "react";
+import { ToggleOrderProps } from "../../type";
+
+export function ToggleOrder({
+  ascending,
+  setAscending,
+  playHistory,
+  setPlayHistory,
+}: ToggleOrderProps) {
+  const [order, setOrder] = useState<string>("手順:昇順");
+
+  function handleOrder(): void {
+    const newOrder = !ascending;
+    setAscending(newOrder);
+
+    if (newOrder) {
+      setOrder("手順:昇順");
+    } else {
+      setOrder("手順:降順");
+    }
+
+    const newPlayHistory = playHistory.slice().reverse();
+    setPlayHistory(newPlayHistory);
+  }
+
+  return <button onClick={() => handleOrder()}>{order}</button>;
```

Comment:
```suggestion
  return <button onClick={handleOrder}>{order}</button>;
```
関数を別関数にラップする必要なさそうです！

Reply:
意味を混同していました・・・

下記のように理解を再整理し、{handleOrder}が適切であると理解をしなおしました・・・！
~~~tsx
return(
<>
    <p onClick={function}></p> //クリック時にfunctionを呼び出す
    <p onClick={function()}></p> //レンダリング時にfunctionを呼び出す
    <p onClick={()=>function}></p> //クリック時にfunctionを呼び出す
    <p onClick={()=>function()}></p> //クリック時にfunctionを呼び出す

    //関数が引数を受け取る必要がある場合は下記を使う
    <p onClick={function(value)}></p> //レンダリング時にfunctionを呼び出す
    <p onClick={()=>function(value)}></p> //クリック時にfunctionを呼び出す
</>
);
~~~